beater/cloudbeat.go

```diff
@@ -287,3 +301,25 @@ func (bt *cloudbeat) Stop() {
 func (bt *cloudbeat) configureProcessors(processorsList processors.PluginConfig) (procs *processors.Processors, err error) {
 	return processors.New(processorsList)
 }
+
+func pipelineStep[In any, Out any](ctx context.Context, inputChannel chan In, fn func(context.Context, In) Out) chan Out {
+	outputCh := make(chan Out)
+
+	go func() {
+		defer close(outputCh)
+
+		for s := range inputChannel {
+			select {
+			case <-ctx.Done():
+				break
+			default:
+			}
+
+			go func(ctx context.Context, s In) {
+				outputCh <- fn(ctx, s)
```

Comment:
It might be confusing but the function actually gets the concrete argument of type `In` and not `channel In`.
Do you still think this design needs to be modified?

Reply:
Hmm my mistake.
In this case it's closer to option 2.
wdyt? Is this worth moving to another module?

Reply:
And also now I see what you met with concurrency.
Current implementation starts n (2) go routines per resource

Reply:
Yeah, I'll create a new module.
It's not really related to the `beater` module.